python/hsfs/core/opensearch.py

```diff
@@ -0,0 +1,79 @@
+#
+#   Copyright 2023 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from hsfs import client
+from hsfs.client.external import Client
+from hsfs.client.exceptions import FeatureStoreException
+import logging
+
+
+class OpenSearchClientSingleton:
+    _instance = None
+
+    def __new__(cls):
+        if not cls._instance:
+            cls._instance = super(OpenSearchClientSingleton, cls).__new__(cls)
+            cls._instance._opensearch_client = None
+            cls._instance._setup_opensearch_client()
+        return cls._instance
+
+    def _setup_opensearch_client(self):
+        if not self._opensearch_client:
+            try:
+                import hopsworks
```

Comment:
We should not use hopsworks library inside hsfs here. Especially since I see we do hopsworks.login() and this is guaranteed to create issues given the complexity of the connection code between hopsworks/hsfs/hsml. This also means the opensearch code being used here needs to be moved to hsfs.

Reply:
isn't the purpose of hopsworks-api is to gather all the common code e.g opensearch, kafka? if so then here it needs to import hopsworks. Not sure what issues will be created regarding the login, but here it can just throw an error and ask the users to login hopsworks by themselves, but essential what they will do is the code here.

Reply:
The purpose of hopsworks is that it will eventually absorb hsfs and hsml. It should not be used from hsfs or hsml directly. It currently provides the functionality that are not in hsfs/hsml because it provides APIs to general services like jobs, kafka, python etc.

Reply:
Yes until we have a solution to manage this. @javierdlrm was working on a design doc at some point for this if you want more info.

Reply:
ok then I will duplicate the opensearch-api related code to hsfs.

Reply:
@robzor92 removed the hopsworks dependency: https://github.com/logicalclocks/feature-store-api/pull/1139/commits/221dad3506f817f508f6950d953f68024207f345